bot/utils/members.py

```diff
@@ -0,0 +1,47 @@
+import logging
+import typing as t
+
+import discord
+
+log = logging.getLogger(__name__)
+
+
+async def get_or_fetch_member(guild: discord.Guild, member_id: int) -> t.Optional[discord.Member]:
+    """
+    Attempt to get a member from cache; on failure fetch from the API.
+
+    Return `None` to indicate the member could not be found.
+    """
+    if member := guild.get_member(member_id):
+        log.trace("%s retrieved from cache.", member)
+    else:
+        try:
+            member = await guild.fetch_member(member_id)
+        except discord.errors.NotFound:
+            log.trace("Failed to fetch %d from API.", member_id)
```

Comment:
That probably should at least be an info log. Same for the other one.

Reply:
Logging for not found members is up to the caller, I didn't want to make this default behaviour as it could be quite spammy depending on where it is called from.

This is copied directly from bot.